Assignment_05.rb

```diff
@@ -72,19 +72,92 @@ def get_time_duration_in_minute(start_time, end_time)
     duration.to_f * 24 * 60
   end
 
+  # This method validate the the id
+  def validates_id
+    if id.nil?
+      puts "Task - validates_id failed - A Task must contain an id"
+      return false
+    end
+
+    unless id.kind_of?(Integer)
+      puts "Task - validates_id failed - id must be Integer"
+      return false
+    end
+
+    true
+  end
+  
+  # This method validate the the name
+  def validates_name
+    if name.nil?
+      puts "Task - validates_name failed - A Task must contain a name"
+      return false
+    end
+
+    unless name.kind_of?(String)
+      puts "Task - validates_name failed - name must be String"
+      return false
+    end
+
+    true
+  end
+
   # This method validate the the time logs
-  def validate_time_logs
+  def validates_time_logs
+    if time_logs.nil?
+      puts "Task - validates_time_logs failed - A Task must contain time_logs"
+      return false
+    end
+
+    unless time_logs.kind_of?(Array)
+      puts "Task - validates_time_logs failed - time_logs must be array"
+      return false
+    end
+
     time_logs.each do |time_log|
-      if DateTime.parse(time_log[0]) > DateTime.parse(time_log[1])
-        puts "Time logs validation failed - end time earlier than start time - for id = #{id} and name = #{name}"
-        return false
-      elsif DateTime.parse(time_log[0]) == DateTime.parse(time_log[1])
-        puts "Time logs validation failed - start time and end time can't be same - for id = #{id} and name = #{name}"
-        return false
-      end
+      return false unless validates_time_log(time_log)
     end
+
     true
   end
+
+  # This method validate the the time log
+  def validates_time_log(time_log)
+    unless time_log.kind_of?(Array)
+      puts "Task - validates_time_log failed - Each time log inside time_logs must be an array"
+      return false
+    end
+
+    unless time_log.length == 2
+      puts "Task - validates_time_log failed - Each time log inside time_logs must be an array of size two"
+      return false
+    end
+
+    unless time_log[0].kind_of?(String) || time_log[1].kind_of?(String)
+      puts "Task - validates_time_log failed - Each time log inside time logs must be an array of size two of type string"
+      return false
+    end
+
+    unless is_valid_date_time_format(time_log[0]) && is_valid_date_time_format(time_log[1])
+      puts "Task - validates_time_log failed - Each time log inside time logs must be in YYYY-MM-DD hh:mm:ss format"
```

Comment:
@AritraChatterjee-qi Can you please re-phrase the error message like `start time and end time` instead of `time log inside time logs`